src/backend/gpopt/utils/COptTasks.cpp

```diff
@@ -498,7 +498,9 @@ COptTasks::OptimizeTask(void *ptr)
 		CMDCache::Init();
 		CMDCache::SetCacheQuota(optimizer_mdcache_size * 1024L);
 	}
-	else if (reset_mdcache)
+	else if (reset_mdcache ||
+			 (gpdb::GPDBTransactionIdIsValid(CMDCache::GetTransientXmin()) &&
```

Comment:
Can we move conditions on 502 and 503 lines to gpdb::MDCacheNeedsReset function?